apps/src/lib/levelbuilder/script-editor/RemoveLessonDialog.jsx

```diff
@@ -0,0 +1,58 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {removeLesson} from '@cdo/apps/lib/levelbuilder/script-editor/scriptEditorRedux';
+import Dialog from '@cdo/apps/templates/Dialog';
+
+/**
+ * Dialog which confirms removal of the lesson in the specified position
+ * within the lesson group.
+ */
+export class UnconnectedRemoveLessonDialog extends Component {
+  static propTypes = {
+    lessonGroupPosition: PropTypes.number.isRequired,
+    // Position of lesson to remove. Dialog opens when this is set.
+    lessonPosToRemove: PropTypes.number,
+    lessonName: PropTypes.string,
+    handleClose: PropTypes.func.isRequired,
+
+    // provided by redux
+    removeLesson: PropTypes.func.isRequired
+  };
+
+  handleConfirm = () => {
+    const {
+      lessonGroupPosition,
+      lessonPosToRemove,
+      removeLesson,
+      handleClose
+    } = this.props;
+    removeLesson(lessonGroupPosition, lessonPosToRemove);
+    handleClose();
+  };
+
+  render() {
+    const {lessonName, handleClose, lessonPosToRemove} = this.props;
+    let bodyText = `Are you sure you want to remove the level named "${lessonName}" from the script?`;
```

Comment:
Sorry to nit, but there is a lot at stake if the user sees this warning, soI think this could benefit from a stronger warning message, like: 

big warning: "Are you sure you want to permanently delete the lesson named X and all its contents?" 

optional smaller text: "This will delete any details within the lesson, such as level progressions and lesson plan markdown. Individual levels will not be deleted."

Reply:
<img width="523" alt="Screen Shot 2020-11-03 at 4 02 38 PM" src="https://user-images.githubusercontent.com/208083/98040288-021fea00-1dee-11eb-80cf-6e2db854eb65.png">

@davidsbailey how is this?